curvefs/test/mds/mds_service_test.cpp

```diff
@@ -151,332 +314,198 @@ class MdsServiceTest : public ::testing::Test {
     }
 
     std::shared_ptr<FsManager> fsManager_;
+    std::shared_ptr<MockMemoryFsStorage> mockFsStorage_;
```

Comment:
it's unnecessary to use shared pointer in here, just `MockMemoryFsStorage mockFsStorage_`

Reply:
@wu-hanqing, it's not possible. We have to assign the value separately in the Setup because the = operator has been deleted in the ancestor class, FsStorage.

(`MockMemoryFsStorage -> MemoryFsStorage -> FsStorage`)

Reply:
I also want to know why not to use `shared_ptr` here, as it provides a safer way to manage this pointer.

Reply:
> @wu-hanqing
> 
> I could be wrong, but I think it is not possible to get rid of the shared pointer. If we directly declare the variable as the member variable, we have to assign the value separately in `Setup`, but the `operator=` is deleted in `FsStorage` (ancestor of `MockMemoryFsStorage `), causing initialization to fail.
> 
> (`MockMemoryFsStorage -> MemoryFsStorage -> FsStorage`)

The reason why you have to assign a value to `mockFsStorage_` is the object doesn't exist before you create one, because it's a NULL pointer.

So, after modification, it's unnecessary to assign again.

Reply:
> I am also curious about why we're not using `shared_ptr` here, as I have learned it is a modern way of managing pointers.

I think `shared_ptr` is the last option to choose when you want to manage pointers.

Not everything needs to be a pointer, prefer bare objects over pointers, and if `unique_ptr` is suitable for use cases, then there's no need to use `shared_ptr`.

Here are some materials that may help you better understand how to create/manage object
- https://abseil.io/tips/123
- https://www.youtube.com/watch?v=xGDLkt-jBJ4

Reply:
Got it.
Thank you for the materials!